### PR TITLE
Jiemingy patch 7

### PR DESCRIPTION
**Pull request by Jiemingy**
Dear John,

I have completed Project 2. The attached patch-7 is my work on it. Hope you can review and correct it if there is any error.

Bests,
Jieming

### CODE REVIEW

File: Project-2/README.md [hunk 1]
```diff
@@ -10,42 +10,261 @@ Tip #4: Work together as a team.
...
+One day, his father gives him a game, called Color-card: there are a number of color cards in a box. Each card has a different color, but it is full colored on one side, while on the other side it is a blank sheet with the colored dot. Once a time, the player chooses any number of cards from the box, and then use any number of cards he chose to make combinations. For example, the first time, Bob chooses five color cards from the box, red, yellow, blue, green, brown, then he uses three cards, and put them in such a order: 1, red (front side), 2, blue (back side), 3, green (back side). That is one combination. The player continues playing until there is a combination that repeats. If the player cannot continue playing, he/she fails. 
+However, if the player fails at one time, he/she can choose more or/and different cards, then play the next time. The player wins if the cards he chooses can allow him to play forever. 
+Unfortunately, Bob cannot win the game, because no matter how many cards he chooses (the number of cards is not limitless), and how hard he tries, he cannot continue playing without a repeated combination occurs.
```
Comment (johnbeve) [inline Project-2/README.md:247]: I like this story, but it's not clear to me how it elucidates the finite model property, other than by saying this card game has it. Examples of formal structures that have the property are of course helpful when ELI5, but we should include a description of the features of the example that are generalizable wrt finite model property.

File: Project-2/README.md [hunk 2]
```diff
@@ -10,42 +10,261 @@ Tip #4: Work together as a team.
...
+We say that the Color-card game has finite model property. 
+Finite model property is important because it can be used to design a decidable algorithm for the satisfiability of its concepts. 
+
 8. Following up on the preceding , explain the _tree model property_. Be sure to provide a simple example and explain when the property might be important, and when it is not so important. 
 
+At another day, Bob's father gives him a different game, called Dot-and-Line. 
+There is a big board, a number of tacks and thread. The games can be played with two people. When it is your turn, the other player choose a number of tacks and tacks them on the board - he/she can choose any place to locate a tack. Then you use thread to connect the tacks (any one tack needs to be connected to at least one other tack) - have all tacks to be connected. The connection is such that from one tack to any other tack through the thread, there is only one way to go. If the other player finds there is more than two ways between any two tacks in your work, you lose. If you does not lose, then it is the other player's turn. You two play in turn until one loses.
```
Comment (johnbeve) [inline Project-2/README.md:254]: Same comment as above, love the story, connect to generalizable features of the tree model property.

**Review by johnbeve — CHANGES_REQUESTED** [review]

**Review by emmoramor — COMMENTED** [review]
Dear Jieming,

Many thanks for the opportunity to review your project, and my apologies that it took me this long to get to this review. Since I am a little late to this review-party, I'll try and be brief/concise. Your answers look good overall, and I second (most of) the other reviewers' comments. I particularly enjoyed your answers to questions 4 and 5.  I do have a couple observations on a couple of your answers. I think your ELI5s could be reworked in some parts to make them accessible to the five year old audience. I explain this in some detail below.

Regarding your answer to question 7, I think it could be made clearer and more accessible for a 5 year old if you rephrase some words and concepts that are either technical, complex, or containing jargon. For example, when you say that fmp is important "because it can be used to design a decidable algorithm for the satisfiability of concepts," this seems too sophisticated for any five year old.  I wonder if you can find a way to convey some of the same insights/abstractions, without sacrificing meaning, by rephrasing this into simpler and more accessible language. How would you explain what an 'algorithm' is to a five year old?

Regarding your answer to question 8... you wrote, "We say that the Dot-and-Line game has a tree model property." I am wondering about the reasoning behind this assertion. Why and how does the Dot-and-Line game have the tmp?  (Note, I am certainly not disagreeing here, but seeking to understand  more clearly where the assertion came from and what justifies/grounds it.)

Regarding your answer to question 9... your last sentence caught my eye. You wrote, "This will add a new fact, that c overlaps a." I am not sure that this is, strictly speaking, a 'new' fact or that it gets 'added' into the picture as 'new' per se. It seems to me that the fact in question, once the relevant assertions are made and entered into the system, was always there--but only implicitly. So, it would be an 'old' fact that is now recognizable. This, then, is precisely the reasoner's task: to bring out these implicit relations that might otherwise go unnoticed, and make them explicit. The fact or inferential relation would be 'new' only in its explicit status, that is, in the sense that previously hidden inferences are now accessible in explicit form. So, it was always the case (once the proper inputs/assertions took place) that, e.g., 'c overlaps a.' Now we just can't miss it.

File: Project-2/README.md (hunk 2 above)
Comment (Jiemingy) [reply Project-2/README.md:254]: Dear John,

Thank you for the comments! I will add the generalizable features of both properties.

Bests,
Jieming

**Comment by Jiemingy** [comment]
Dear John,
I have added the generalization features of the finite model property and tree model property, and also explanations of why the game has one of the properties. 
Could you review the added part?
Bests,
Jieming
